pkg/client/client_test.go

```diff
@@ -106,12 +172,22 @@ func TestClient(t *testing.T) {
 			if h.r.URL.String() != expURL {
 				t.Fatalf("should have %s for URL, got: %s", expURL, h.r.URL.String())
 			}
+
+			if _, _, ok := h.r.BasicAuth(); ok {
+				t.Fatalf("basic auth should not be set")
+			}
+
+			payload := getPayload(h)
+
+			if payload.ClientParams.Group != test.expCfg.Group {
+				t.Fatalf("payload's group should be : %s, got: %s", test.expCfg.Group, payload.ClientParams.Group)
+			}
```

Comment:
I like checking the payload. However, if we do that, should we also verify the headers?

Reply:
To assert this: https://github.com/flatcar-linux/fleetlock/blob/main/pkg/client/client.go#L117 ?

Reply:
https://github.com/flatcar-linux/fleetlock/blob/8937d308dfa43a7f4ec308005c9b75c569e0e79a/pkg/client/client.go#L117-L118 to be precise, yes :+1: